include/ibamr/IBHydrodynamicForceEvaluator.h

```diff
@@ -0,0 +1,339 @@
+// modification, are permitted provided that the following conditions are met:
+//
+//    * Redistributions of source code must retain the above copyright notice,
+//      this list of conditions and the following disclaimer.
+//
+//    * Redistributions in binary form must reproduce the above copyright
+//      notice, this list of conditions and the following disclaimer in the
+//      documentation and/or other materials provided with the distribution.
+//
+//    * Neither the name of The University of North Carolina nor the names of
+//      its contributors may be used to endorse or promote products derived from
+//      this software without specific prior written permission.
+//
+// THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+// AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+// IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+// ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE
+// LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+// CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+// SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+// INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+// CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+// ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+// POSSIBILITY OF SUCH DAMAGE.
+
+#ifndef included_IBHydrodynamicForceEvaluator
+#define included_IBHydrodynamicForceEvaluator
+
+#include <map>
+#include <vector>
+
+#include "Box.h"
+#include "Eigen/Core"
+#include "Eigen/Geometry"
+#include "tbox/Serializable.h"
+#include <ibtk/LData.h>
+#include <ibtk/LDataManager.h>
+
+namespace IBTK
+{
+class LData;
+} // namespace IBTK
+namespace SAMRAI
+{
+namespace hier
+{
+template <int DIM>
+class PatchHierarchy;
+} // namespace hier
+namespace tbox
+{
+template <class TYPE>
+class Pointer;
+} // namespace tbox
+} // namespace SAMRAI
+
+/////////////////////////////// CLASS DEFINITION /////////////////////////////
+
+namespace IBAMR
+{
+/*!
+ * \brief Class IBHydrodynamicForceEvaluator computes hydrodynamic force and
+ * torque on immersed bodies. The class uses Reynolds transport theorem to integrate
+ * momentum over a Cartesian box region that moves with an arbitrary rigid body
+ * translation velocity.
+ *
+ * References
+ * Flavio Noca, <A HREF="http://thesis.library.caltech.edu/3081/1/Noca_f_1997.pdf">On the evaluation
+ * of time-dependent fluid-dynamic forces on bluff bodies.</A>
+ *
+ * Nangia et al. A moving control volume approach to computing hydrodynamic forces and
+ * torques on immersed bodies.
+ *
+ * \note  The Cartesian box should enclose the body entirely.
+ * \note  Various IB methods need to provide linear and angular momentum of the
+ *  enclosed body to the class.
+ */
+class IBHydrodynamicForceEvaluator : public SAMRAI::tbox::Serializable
```

Comment:
Is there crucial object state that requires serialization? If not, I would suggest dropping the Serializable interface here.